Tests/SonarScanner.MSBuild.Common.Test/ProcessRunnerTests.cs

```diff
@@ -479,36 +429,87 @@ private static void SafeSetEnvironmentVariable(string key, string value, Environ
         }
         catch (SecurityException)
         {
-            logger.LogWarning("Test setup error: user running the test doesn't have the permissions to set the environment variable. Key: {0}, value: {1}, target: {2}",
-                key, value, target);
+            logger.LogWarning(
+                "Test setup error: user running the test doesn't have the permissions to set the environment variable. Key: {0}, value: {1}, target: {2}",
+                key,
+                value,
+                target);
         }
     }
 
     private static string LogArgsPath() =>
         // Replace to change this project directory to LogArgs project directory while keeping the same build configuration (Debug/Release)
         Path.Combine(Path.GetDirectoryName(typeof(ProcessRunnerTests).Assembly.Location).Replace("SonarScanner.MSBuild.Common.Test", "LogArgs"), "LogArgs.exe");
 
-    private void AssertExpectedLogContents(string logDir, params string[] expected)
+    private class ProcessRunnerContext
     {
-        var logFile = Path.Combine(logDir, "LogArgs.log");
-        File.Exists(logFile).Should().BeTrue("Expecting the argument log file to exist. File: {0}", logFile);
-        TestContext.AddResultFile(logFile);
-        var actualArgs = File.ReadAllLines(logFile);
-        actualArgs.Should().BeEquivalentTo(expected, "Log file does not have the expected content");
-    }
+        private readonly ProcessRunner runner;
+        private readonly string testDir;
+        private readonly string exePath;
+        private ProcessResult result;
 
-    private static void AssertTextDoesNotAppearInLog(string text, TestLogger logger)
-    {
-        AssertTextDoesNotAppearInLog(text, logger.InfoMessages);
-        AssertTextDoesNotAppearInLog(text, logger.Errors);
-        AssertTextDoesNotAppearInLog(text, logger.Warnings);
-    }
+        public TestLogger Logger { get; private set; }
+        public int ExpectedExitCode { get; init; }
+        public ProcessRunnerArguments ProcessArgs { get; init; }
 
-    private static void AssertTextDoesNotAppearInLog(string text, IList<string> logEntries)
-    {
-        logEntries.Should().NotContain(x => x.IndexOf(text, StringComparison.OrdinalIgnoreCase) > -1,
-            "Specified text should not appear anywhere in the log file: {0}", text);
-    }
+        public ProcessRunnerContext(TestContext testContext, string commands)
```

Comment:
```suggestion
        public ProcessRunnerContext(TestContext testContext, string commands = null)
```
From what I can tell, `null` and `string.Empty` have the same behavior, so this will simplify a bunch of calls.

Reply:
oh i cant :D